core/src/ncipher/aes_gcm.c

```diff
@@ -26,6 +26,13 @@ uint8_t aes_gcm_buffer[1000];
 Result aes_gcm_encrypt(M_KeyID keyId, uint8_t *plaintext, size_t plaintext_len,
                        uint8_t *ciphertext, size_t ciphertext_len,
                        size_t *bytes_written) {
+
+  // NULL pointer checks. Note that plaintext can be NULL.
```

Comment:
The current callers of this function in subzero do not provide a NULL plaintext. However, because this pair of functions can potentially be used in wider use case scenarios, and GCM allows empty plaintext (i.e., authentication-only), it's worthwhile to be accurate in these cases, to avoid issues in future use cases.

Reply:
I would reject a null plaintext. If you want to process an empty plaintext, you can simply call the function with an empty string.

Reply:
I like the suggestion. Although the underlying GCM implementation (in the 'dev' case at least) takes a NULL plaintext pointer as a valid one, it does make it simpler to reject this case in this layer.

Reply:
whatever behavior we settle on should be clearly/succinctly documented in the function documentation. +1 for the simplest possible behavior (banning null plaintext). it would also simplify the explicit handling of NULL `plaintext` below/